### DISCUSSION TITLE
Opinion on microstream

### DISCUSSION THREAD

**Discussion by guykatz** [quarkusio/quarkus#30103] [Community]
Hi all
Usually, every interesting tech I bump into i can easily find info on quarkus integ with it.
Recently I gained interest in microstream. It seems like it's gaining a lot of momentum, already supported by hellidon and micronaut which in my eyes are the direct competition to quarkus.
In this aspect, quarkus is a bit behind and is very silent about microstream  I was wondering, does quarkus have any plan to provide tight integration with this library?I would live to get your expert opinion on this.
I would def b interested in getting rid of my db
Your input is welcomed...

**Comment by maxandersen** [comment]
There are already some who made a Quarkus extension for Microstream - see https://github.com/belu/microquark

That is the wonderful part of the Quarkus extension model - anyone can come and provide direct integration.

About expert opinion - then I'll just comment that microsteam does not replace your database. It gives you another database using an approach that is closer to serialization than relational or even document based database. Such an approach can be useful for some; just be careful and be aware of what you are stepping into - persistence is not trivial - don't blindly trust any marketing (including Quarkus), verify for yourself and your usecase.

**Reply by guykatz** [reply]
Thanks @maxandersen 
Maybe this tech is worth an insights episode? There are so many persistence options it's harder and harder to choose...

**Comment by sorin-costea** [comment]
A bit late but in case anyone is looking, the Microstream Quarkus extension is explained here:
https://github.com/rdebusscher/microstream-quarkus-patterns

**Reply by guykatz** [reply]
thanks I have seen this in the release notes of microstream.
like the tech -> started the port to it as it fits my usecases.
quarkus is on hold for me at the moment but I am keeping a close eye on it :)

**Comment by guykatz** [comment]
@sorin-costea  also please note the specific Quarkus integration in the Docs: [https://docs.microstream.one/manual/misc/integrations/quarkus.html](https://docs.microstream.one/manual/misc/integrations/quarkus.html)
